app/models/team.rb

```diff
@@ -3,4 +3,59 @@ class Team < ApplicationRecord
   has_many :models, dependent: :nullify
 
   validates :name, presence: true, uniqueness: true
+
+  scope :models, (lambda do
+    joins('LEFT JOIN "models"
+      ON "models"."team_id" = "teams"."id"').
+      group('teams.id')
+  end)
+
+  scope :users, (lambda do
+    joins('LEFT JOIN "users"
+      ON "users"."team_id" = "teams"."id"').
+      group('teams.id')
+  end)
+
+  ORDERS = %w[name models members].freeze
+
+  class << self
+    def fetch_all(options)
+      teams = Team
+      options.each_with_index do |filter|
+        teams = apply_filter(teams, options, filter[0], filter[1])
+      end
+      teams = teams.order(name: :asc) unless options['order_type'].present?
+      teams
+    end
+
+    def apply_filter(teams, options, filter, value)
+      case filter
+      when 'search'
+        teams.where("lower(teams.name) LIKE '%#{value.downcase}%'")
```

Comment:
actually I think we have the same issue in Indicator & Scenario as well, so maybe let's apply the fix across the board :)